doc/visual-programming/source/widgets/model/stochasticgradient.rst

```diff
@@ -0,0 +1,103 @@
+Stochastic Gradient Descent
+===========================
+
+.. figure:: icons/stochastic-gradient.png
+
+Minimize an objective function using a stochastic approximation of gradient descent.
+
+Signals
+-------
+
+**Inputs**:
+
+-  **Data**
+
+   A data set.
+
+-  **Preprocessor**
+
+   Preprocessing method(s)
+
+**Outputs**:
+
+-  **Learner**
+
+   A SGD regression learning algorithm with settings as specified in the dialog.
+
+-  **Model**
+
+   A trained model. Output signal sent only if input *Data* is present.
+
+Description
+-----------
+
+The **Stochastic Gradient Descent** widget uses `stochastic gradient descent <https://en.wikipedia.org/wiki/Stochastic_gradient_descent>`_ that minimizes a chosen loss function with a linear function. The algorithm approximates a true gradient by considering one sample at a time, and simultaneously updates the model based on the gradient of the loss function. For regression, it returns predictors as minimizers of the sum, i.e. M-estimators, and is especially useful for large-scale and sparse data sets.
+
+.. figure:: images/StochasticGradientDescent-stamped.png
+   :scale: 50 %
+
+1. Specify the name of the model. The default name is "SGD".
+2. Algorithm parameters. Classification loss function:
+
+   -  `Hinge <>`_ ()
```

Comment:
No, SGD is just the optimization method. If we optimize a hinge loss with SGD, we get a linear SVM, we've just gotten to it using a different method (I'm guessing the regular SVM in sklearn uses lbfgs or some other super optimized method). Similarly, if we optimize the log loss, we get logistic regression or squared loss for regression, we get linear regression. This is only ever really useful when we have a lot of data.

Imo, it's kind of a strange separation on scikit-learn's part... It's also strange they don't implement mini-batch gradient descent if they've already gone with sgd...